lib/src/bitarray.dart

```diff
@@ -0,0 +1,317 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection' show ListMixin;
+import 'dart:typed_data' show Uint32List;
+import 'package:collection/collection.dart' show NonGrowableListMixin;
+
+/// A BitArray implementation for Dart to store boolean values.
+///
+/// Uses list of integers as internal storage to reduce memory usage.
+/// Implemented as fixed-length [List] of [bool].
+/// Throws error when calling any operation with length modifying.
```

Comment:
Every time we create a fixed-length list class, at some point in the future someone needs the growable version and writes a completely new implementation. Almost all the code is the same, but duplicated.
I would suggest that this library embrace this future and build on a base class that can be used for both fixed-length and growable implementations.